app/src/main/java/org/p2p/wallet/moonpay/model/SellTransaction.kt

```diff
@@ -51,5 +52,14 @@ sealed class SellTransaction(
         override val amounts: SellTransactionAmounts,
         override val selectedFiat: SellTransactionFiatCurrency,
         override val userAddress: Base58String,
+        val failureReason: SellTransactionFailureReason?,
     ) : SellTransaction(SellTransactionStatus.FAILED)
 }
+
+fun SellTransaction.isCancelled(): Boolean {
+    return if (this is SellTransaction.FailedTransaction) {
+        this.failureReason != SellTransactionFailureReason.CANCELLED
+    } else {
+        true
+    }
```

Comment:
`this is SellTransaction.FailedTransaction && failureReason != SellTransactionFailureReason.CANCELLED`